### PR TITLE
Added JES boot disk runtime option Closes #590

### LINKED ISSUE #590: Add support for JES boot disk

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Comment by cjllanwarne** [comment]
@kshakir @Horneth please!

**Comment by kshakir** [comment]
Hashes are busted in travis for some reason, but overall :+1:

**Comment by cjllanwarne** [comment]
Since @Horneth is away, I'll respin... @mcovarr if that's ok?

**Comment by mcovarr** [comment]
:sob: 

I mean, sure.

File: engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala [hunk 1]
```diff
@@ -225,6 +229,16 @@ object CromwellRuntimeAttributes {
     }
   }
 
+  private def validateBootDisk(diskSize: Option[WdlValue]): ErrorOr[Int] = diskSize match {
+    case Some(x: WdlValue) if WdlIntegerType.isCoerceableFrom(x.wdlType) =>
+      WdlIntegerType.coerceRawValue(x) match {
+        case scala.util.Success(x: WdlInteger) => x.value.intValue.successNel
+        case scala.util.Success(coersionhasgoneawry) => s"Coercion was expected to create an Integer but instead got $coersionhasgoneawry".failureNel
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: pretty sure that's not a correct spelling of "coercion" even in British

File: engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala [hunk 2]
```diff
@@ -225,6 +229,16 @@ object CromwellRuntimeAttributes {
     }
   }
 
+  private def validateBootDisk(diskSize: Option[WdlValue]): ErrorOr[Int] = diskSize match {
+    case Some(x: WdlValue) if WdlIntegerType.isCoerceableFrom(x.wdlType) =>
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: do you need the `WdlValue` ascription?

File: engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala [hunk 3]
```diff
@@ -92,7 +93,8 @@ object CromwellRuntimeAttributes {
                         ValidKeyType("continueOnReturnCode", Set(WdlBooleanType, WdlArrayType(WdlIntegerType))),
                         ValidKeyType("failOnStderr", Set(WdlBooleanType)),
                         ValidKeyType("preemptible", Set(WdlIntegerType)),
-                        ValidKeyType("memory", Set(WdlStringType)))
+                        ValidKeyType("memory", Set(WdlStringType)),
+                        ValidKeyType("boot_disk", Set(WdlStringType)))
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: why is this not `bootDiskSizeGb`?

File: engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala (hunk 1 above)
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: maybe print the type that popped out too?

File: engine/src/test/scala/cromwell/engine/backend/jes/JesRuntimeInfoSpec.scala [hunk 4]
```diff
@@ -39,7 +40,8 @@ class JesRuntimeInfoSpec extends FlatSpec with Matchers {
       // NOTE: This value is irrelevant for this test as it is not sufficient to determine if a Call should be started with a preemptible VM or not
       preemptible = 3,
       disks = disks,
-      memoryGB = 4D)
+      memoryGB = 4D,
```
Comment (mcovarr) [inline engine/src/test/scala/cromwell/engine/backend/jes/JesRuntimeInfoSpec.scala]: why `D`?

File: engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala (hunk 1 above)
Comment (geoffjentry) [inline engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: @mcovarr we need an execusion engine for that

Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: yeah we can localize Cromwell to en_WRONG

**Comment by mcovarr** [comment]
Travis thinks you forgot to update some hash expectations

File: engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala (hunk 1 above)
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: `toString` includes the type IIRC. Only `valueString` removes it.

File: engine/src/test/scala/cromwell/engine/backend/jes/JesRuntimeInfoSpec.scala (hunk 4 above)
Comment (cjllanwarne) [inline engine/src/test/scala/cromwell/engine/backend/jes/JesRuntimeInfoSpec.scala]: Shrug? I just removed the comma

**Comment by cjllanwarne** [comment]
@mcovarr Fixes made. Not sure what's up with Travis now. The tests themselves passed, assembly failed for deduplication reasons...?

**Comment by mcovarr** [comment]
The assembly works locally?

**Comment by cjllanwarne** [comment]
@mcovarr seems to be holding together now...

File: project/plugins.sbt [hunk 5]
```diff
@@ -5,4 +5,4 @@ addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "0.8.5")
 addSbtPlugin("com.github.gseitz" % "sbt-release" % "0.8.3")
 addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.3.5")
 addSbtPlugin("org.scoverage" % "sbt-coveralls" % "1.1.0")
-
+addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.8.2")
```
Comment (mcovarr) [inline project/plugins.sbt]: do we really want to add this?

File: project/Dependencies.scala [hunk 6]
```diff
@@ -41,8 +41,8 @@ object Dependencies {
     "com.google.api-client" % "google-api-client-java6" % googleClientApiV,
     "com.google.api-client" % "google-api-client-jackson2" % googleClientApiV,
     "com.google.oauth-client" % "google-oauth-client" % googleClientApiV,
-    "com.google.cloud.bigdataoss" % "gcsio" % "1.4.3",
-    "com.google.apis" % "google-api-services-genomics" % "v1alpha2-rev7-1.19.1"
+    "com.google.cloud.bigdataoss" % "gcsio" % "1.4.4",
+    "com.google.apis" % "google-api-services-genomics" % ("v1alpha2-rev14-" + googleClientApiV) //1.21.0"
```
Comment (mcovarr) [inline project/Dependencies.scala]: the commented out part dangling off the end should go

**Comment by mcovarr** [comment]
:+1: assuming Travis concurs

**Pull request merged**